sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSourceSuite.scala

```diff
@@ -1059,6 +1060,72 @@ class FileStreamSourceSuite extends FileStreamSourceTest {
     val str = Source.fromFile(getClass.getResource(s"/structured-streaming/$file").toURI).mkString
     SerializedOffset(str.trim)
   }
+
+  test("FileStreamSource - latestFirst") {
+    withTempDir { src =>
+      // Prepare two files: 1.txt, 2.txt, and make sure they have different modified time.
+      val f1 = stringToFile(new File(src, "1.txt"), "1")
+      val f2 = stringToFile(new File(src, "2.txt"), "2")
+      eventually(timeout(streamingTimeout)) {
+        f2.setLastModified(System.currentTimeMillis())
+        assert(f1.lastModified < f2.lastModified)
+      }
+
+      // Read oldest files first, so the first batch is "1", and the second batch is "2".
+      val fileStream = createFileStream(
+        "text",
+        src.getCanonicalPath,
+        options = Map("latestFirst" -> "false", "maxFilesPerTrigger" -> "1"))
+      val clock = new StreamManualClock()
+      testStream(fileStream)(
+        StartStream(trigger = ProcessingTime(10), triggerClock = clock),
+        AssertOnQuery { _ =>
```

Comment:
> why do you need to wait on the manual clock? CheckLastBatch will automatically wait for the batch to complete?

`CheckLastBatch` waits only when `AddData` is used, but in this test, I need to add data before starting the query.